src/dvclive/live.py

```diff
@@ -506,13 +515,23 @@ def log_image(
             raise InvalidImageNameError(name)
 
         if name in self._images:
-            image = self._images[name]
+            image = self._images[name]["image"]
         else:
             image = Image(name, self.plots_dir)
-            self._images[name] = image
+            self._images[name] = {"image": image}
 
         image.step = self.step
         image.dump(val)
+
+        if annotations:
+            if not Annotations.could_log(annotations):
+                raise InvalidDataTypeError(name, type(annotations))
+
+            annotation_object = Annotations(name, self.plots_dir)
+            self._images[name]["annotations"] = annotation_object
+            annotation_object.step = self.step
```

Comment:
Is this used somewhere?

Reply:
No, it is inherited from the parent object so I figured it was nice to keep it updated with `live.step`. This way future developers will have the expected behavior.
Also I follow the pattern found for the image in `log_image`. The `step` value for an image is only used for `send_to_studio` I believe so that we don't send the image twice. So I believe the `annotation.step` is going to be used for the same reason. 

One other solution is to bring "step" at the same level as "image" and "annotations" in the `self._images` variable.
Today it looks like this:
```json
self._images = {"path": {"image": <ImageObject with a step property>, "annotations":  <AnnotationsObject with a step property>}, "path2": ...}
```